File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/LibrarycardloseColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LibraryCardLoseRow } from "./LibraryCardLoseRow";

export interface LibraryCardLoseColumns {
    LibraryCardLostId: Column<LibraryCardLoseRow>;
    LibraryCardNo: Column<LibraryCardLoseRow>;
    OperateUserUsername: Column<LibraryCardLoseRow>;
    CreateTime: Column<LibraryCardLoseRow>;
    UpdateTime: Column<LibraryCardLoseRow>;
}

export class LibraryCardLoseColumns extends ColumnsBase<LibraryCardLoseRow> {
    static readonly columnsKey = 'ReaderManage.LibraryCardLose';
    static readonly Fields = fieldsProxy<LibraryCardLoseColumns>();
}

File: LMIS/LMIS.Web/Modules/ReaderManage/LibraryCardLose/LibraryCardLoseColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.ReaderManage.Columns;

[ColumnsScript("ReaderManage.LibraryCardLose")]
[BasedOnRow(typeof(LibraryCardLoseRow), CheckNames = true)]
public class LibraryCardLoseColumns
{
    [EditLink,AlignRight]
    public long LibraryCardLoseId { get; set; }
    public string LibraryCardNo { get; set; }
    public string OperateUserUsername { get; set; }
    public DateTime CreateTime { get; set; }
    public DateTime UpdateTime { get; set; }
}